File: ucf_pt/local2025QualifyingRoundData/solutions/tourney/tourney.cpp
// Arup Guha
// 8/2/2025
// Alternate Solution to 2025 UCF Locals Qualifying Problem: Single Elimination Tournament

using namespace std;
#include <bits/stdc++.h>

int main() {

    // Read in input.
    int n;
    cin >> n;

    // Get highest one bit...
    int highbit = 0;
    while ( (1<<highbit) <= n ) highbit++;
    highbit--;

    // We find the highest power of 2 <= n.
    // Then subtract this from n to give the of winning teams
    // in the prelims...multiply by 2 to get # of teams in prelims.
    cout << ((n-(1<<highbit))<<1) << endl;

    return 0;
}

File: ucf_pt/local2025QualifyingRoundData/solutions/tourney/tourney.py
# Arup Guha
# Alternate Solution to 2025 UCF Locals Qualifying Problem: Single-Elimination Tournament
# 8/2/2025

# Get input.
n = int(input())

# So, take the difference from the highest power of 2 <= to n
# This is the number of teams that advance from prelim to round 1
# So we just need to double that value (left shift by 1).
print( (n - (1<< (n.bit_length()-1) )) << 1 )
